packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -161,27 +161,28 @@ export class SelectionManager implements MultipleSelectionManager {
    */
   extendSelection(toKey: Key) {
     toKey = this.getKey(toKey);
-    this.state.setSelectedKeys(selectedKeys => {
```

Comment:
was this to deal with the issue that you had where unmounted elements weren't running this state update?
we do use the previous state to determine our next state, was useReducer considered?
I wonder if we could flush the update on unmount somehow manually, though i don't know how we'd do that without a ref that we keep updated with the state and call onChange ourselves with that value :(

Reply:
Yeah I _think_ the problem was that the callback is not guaranteed to be called by React if the element unmounts. This was happening with the menu. I think useReducer would have the same problem (useState is implemented using useReducer internally).

The main thing is I think we don't want to wait to fire `onSelectionChange` until a re-rerender. We were relatively inconsistent about this before: some methods in SelectionManager used the callback like these two, and the others did not. Changing it didn't seem to break anything? Not sure what real world case would break because of this. Any idea?

Reply:
"useState is implemented using useReducer internally" TIL

I can't think of one right now, but still gives me the heebie jeebies. I'll keep thinking.